lib/omnibus/packagers/zip.rb

```diff
@@ -0,0 +1,163 @@
+          if File.directory?(extra_package_dir)
+            # Let's collect the DirectoryRefs
+            zip_source_path += "#{windows_safe_path(extra_package_dir)}\\* "
+          end
+        end
+      end
+      zip_file = windows_safe_path(Config.package_dir, zip_name)
+      zip_source_path += "#{windows_safe_path(project.install_dir)}\\*"
+      cmd = <<-EOH.split.join(" ").squeeze(" ").strip
+      7z a -r
+      #{zip_file}
+      #{zip_source_path}
+      EOH
+      shellout!(cmd)
+
+    end
+
+    #
+    # @!group DSL methods
+    # --------------------------------------------------
+
+    #
+    # set or retrieve additional files to sign
+    #
+    def additional_sign_files(val = NULL)
+      if null?(val)
+        @additional_sign_files
+      else 
+        unless val.is_a?(Array)
+          raise InvalidValue.new(:additional_sign_files, "be an Array")
+        end
+        @additional_sign_files = val
+      end
+    end
+    expose :additional_sign_files
+
+    def extra_package_dirs(val = NULL)
+      if null?(val)
+        @extra_package_dirs || nil
+      else
+        unless val.is_a?(Array)
+          raise InvalidValue.new(:extra_package_dir, "be an Array")
+        end
+        @extra_package_dirs = val
+      end
+    end
+    expose :extra_package_dirs
+
+    #
+    # Discovers a path to a gem/file included in a gem under the install directory.
+    #
+    # @example
+    #   gem_path 'chef-[0-9]*-mingw32' -> 'some/path/to/gems/chef-version-mingw32'
+    #
+    # @param [String] glob
+    #   a ruby acceptable glob path such as with **, *, [] etc.
+    #
+    # @return [String] path relative to the project's install_dir
+    #
+    # Raises exception the glob matches 0 or more than 1 file/directory.
+    #
+    def gem_path(glob = NULL)
+      unless glob.is_a?(String) || null?(glob)
+        raise InvalidValue.new(:glob, "be an String")
+      end
+
+      install_path = Pathname.new(project.install_dir)
+
+      # Find path in which the Chef gem is installed
+      search_pattern = install_path.join("**", "gems")
+      search_pattern = search_pattern.join(glob) unless null?(glob)
+      file_paths = Pathname.glob(search_pattern).find
+
+      raise "Could not find `#{search_pattern}'!" if file_paths.none?
+      raise "Multiple possible matches of `#{search_pattern}'! : #{file_paths}" if file_paths.count > 1
+      file_paths.first.relative_path_from(install_path).to_s
+    end
+    expose :gem_path
+
+    #
+    # @!endgroup
+    # --------------------------------------------------
+
+    # @see Base#package_name
+    def package_name
+      zip_name
+    end
+
+    def zip_name
+      "#{project.package_name}-#{project.build_version}-#{project.build_iteration}-#{Config.windows_arch}.zip"
+    end
+
+    
+    #
+    # The path where the MSI resources will live.
+    #
+    # @return [String]
+    #
+    def resources_dir
+      File.expand_path("#{staging_dir}/Resources")
+    end
```

Comment:
We also don't use the `resources_dir` here. For the `.msi` and `.dmg`, it makes sense as it stores the image and icon files as well as some specification files, but for such a simple packager we can get rid of it (for example, the linux packagers do not have a resources directory).